dev/release/README.md

```diff
@@ -272,6 +272,11 @@ it to GitHub Container Registry at https://github.com/apache/datafusion-comet/pk
 
 Reply to the vote thread to close the vote and announce the release.
 
+## Update released version number in documentation
+
+- We provide direct links to the jar files in Maven
+- The Kubernetes page needs updating once the Docker image has been published to GitHub Container Regsistry
```

Comment:
We probably can mention this PR https://github.com/apache/datafusion-comet/pull/1418 as an example what needs to be changed with the new version? Hope we can get this automated